moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -711,6 +711,18 @@ void PlanningScene::getPlanningSceneDiffMsg(moveit_msgs::PlanningScene& scene_ms
     if (do_omap)
       getOctomapMsg(scene_msg.world.octomap);
   }
+
+  // Ensure any detached collision objects get detached from the parent planning scene, too
+  for (const auto& collision_object : scene_msg.world.collision_objects)
```

Comment:
What happens if the object is completely gone? Ie no longer attached but also removed from world.collision_objects?

Reply:
Do you mean after getting the diff as a message? I think in that case it shouldn't change the previous behavior